plugins/ingestion-kinesis/src/internalClusterTest/java/org/opensearch/plugin/kinesis/IngestFromKinesisIT.java

```diff
@@ -94,8 +94,11 @@ public void testKinesisIngestion() {
 
     public void testKinesisIngestion_RewindByOffset() throws InterruptedException {
         produceData("1", "name1", "24");
-        String sequenceNumber = produceData("2", "name2", "20");
-        Thread.sleep(1000);
+        produceData("2", "name2", "24");
+        String sequenceNumber = produceData("3", "name3", "20");
+        logger.info("Produced message with sequence number: {}", sequenceNumber);
+        produceData("4", "name4", "21");
+        Thread.sleep(2000);
```

Comment:
Can we replace this `Thread.sleep` with a `CountDownLatch` that waits for things to be in the correct place? 

The build servers are often very heavily loaded, so however much time we sleep, it may (sometimes) not be sufficient for the concurrent task to complete. As a bonus, when the build server is not busy, we might wait much less than 2 seconds.

Reply:
sounds good. rewrote it and PTAL